server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java

```diff
@@ -331,56 +327,29 @@ public void checkParentLimit(long newBytesReserved, String label) throws Circuit
                 message.append("]");
             }
             message.append(", usages [");
-            message.append(String.join(", ",
-                this.breakers.entrySet().stream().map(e -> {
+            message.append(this.breakers.entrySet().stream().map(e -> {
                     final CircuitBreaker breaker = e.getValue();
                     final long breakerUsed = (long)(breaker.getUsed() * breaker.getOverhead());
                     return e.getKey() + "=" + breakerUsed + "/" + new ByteSizeValue(breakerUsed);
-                })
-                    .collect(Collectors.toList())));
+                }).collect(Collectors.joining(", ")));
             message.append("]");
             // derive durability of a tripped parent breaker depending on whether the majority of memory tracked by
             // child circuit breakers is categorized as transient or permanent.
             CircuitBreaker.Durability durability = memoryUsed.transientChildUsage >= memoryUsed.permanentChildUsage ?
                 CircuitBreaker.Durability.TRANSIENT : CircuitBreaker.Durability.PERMANENT;
-            logger.debug("{}", message);
+            logger.debug(() -> new ParameterizedMessage("{}", message.toString()));
             throw new CircuitBreakingException(message.toString(), memoryUsed.totalUsage, parentLimit, durability);
         }
     }
 
-    /**
-     * Allows to register a custom circuit breaker.
-     * Warning: Will overwrite any existing custom breaker with the same name.
-     */
-    @Override
-    public void registerBreaker(BreakerSettings breakerSettings) {
+    public CircuitBreaker validateAndCreateBreaker(BreakerSettings breakerSettings) {
```

Comment:
`ChildMemoryCircuitBreaker` objects need a reference to the constructed `HierarchyCircuitBreakerService`. This complicates plugin access to their own created circuit breakers. To prevent muddying the API between the plugin and the node, I have split the breaker creation and the breaker registration.